microservices/payment/src/services/payment-gateway/abstract.ts

```diff
@@ -82,6 +88,33 @@ abstract class Abstract {
 
     return customer;
   }
+
+  /**
+   * Get connect account
+   */
+  protected async getConnectAccount(userId: string) {
+    const connectAccount = await this.connectAccountRepository.findOne(userId);
+
+    if (connectAccount) {
+      return connectAccount;
+    }
+
+    return this.createConnectAccount(userId);
+  }
+
+  /**
+   * Create new connect account
+   */
+  public async createConnectAccount(userId: string, connectAccountId: string = uuid()) {
+    const connectAccount = this.connectAccountRepository.create({
+      connectAccountId,
+      userId,
+    });
+
+    await this.customerRepository.save(connectAccount);
+
+    return connectAccount;
+  }
```

Comment:
move to separated service
microservices/payment/src/services/stripe/connect-account.ts